views/projects.tpl

```diff
@@ -158,125 +159,133 @@
-                                </td>
-
-                                <td class="proj_name dt-nowrap">
-                                    <a href="/showcase/labs/{{ lab_id }}/{{ project_id }}">{{ name }}</a>
-                                </td>
-
-                                % if url:
-                                <td class=""><a href="{{ url }}">Home page</a></td>
-                                % else:
-                                <td class=""></td>
-                                % end
-
-                                <td class="dt-center">{{ date_added.date() }}</td>
-
-                                <td class="dt-center">{{ date_updated.date() }}</td>
-
-                                % maturity_image = {1: 'showcase', 2: 'incubator', 3: 'market'}
-                                <td data-order="{{ maturity + 0.5 if active else maturity }}" class="dt-center">
-                                    <img
-                                        src="/resources/maturity_{{ maturity_image.get(maturity, "na") }}.svg"
-                                        width="25em"
-                                        height="25em"
-                                        title="{{ maturity_label.get(maturity, 0) }}"
-                                        alt="{{ maturity_label.get(maturity, 0) }}"
-                                    >
-                                </td>
-
-                                <td>{{ description }}</td>
-
-                                <td>{{ tech_desc }}</td>
-
-                                <td>{{ layman_desc }}</td>
-
-                                <td class="dt-center">{{ language }}</td>
-
-                                <td>{{ proj_type }}</td>
-
-                                % if 'url' in code:
-                                <td class="dt-nowrap"><a href="{{ code['url'] }}">{{ code.get('type', '') }}</a></td>
-                                % else:
-                                <td>{{ code.get('type', '') }}</td>
-                                % end
-
-                                <td class="dt-center">{{ date_last_commit.date() if date_last_commit else '' }}</td>
-
-                                <td class="dt-center">{{ loc }}</td>
-
-                                <td class="dt-center">
-                                    % if doc:
-                                    <a href="{{ doc }}">link</a>
+                % for category_show in categories.items():
+                    % for lab_id, lab in labs.items():
+                        % if selected_lab_id is None or selected_lab_id == lab_id:
+                            % for project_id, project in lab['projects'].items():
+                                <%
+                                category = ', '.join(map(str, project.get('category', [])))
+                                if category != category_show[0]:
+                                    continue
+                                end
```

Comment:
Since we are sorting on the category column first, I believe we can remove this logic and the surrounding loop, and just display the category in the first column.

Reply:
yes, it was because the `project.get('category')` always returned a `YAML(Other)` (or whatever the category was). And I didn't find how to convert it to a string :(

So now I'm using `map(str, ...)` as a band-aid, but probably that's easier in another way...

Reply:
Yes, converting the `YAML(...)` with `str()` is correct, but I was referring to the loop on the categories and skipping out of the loop when the category does not match.

Reply:
OK, now I see the light. Yes, this comes from my first attempt to do the `colspan` headers.